### PR TITLE
Update official build to generate and upload NuGet packages to the feed

### PR DESCRIPTION
**Pull request by kotlarmilos**
## Description

This PR updates the `prepare-signed-artifacts.yml` file to use Arcade's templates and modifies the publish script to generate and upload NuGet packages to the feed.

Official build: https://dev.azure.com/dnceng/internal/_build/results?buildId=2463272&view=results

### CODE REVIEW

File: eng/pipelines/common/templates/build-job.yml [hunk 1]
```diff
@@ -45,7 +45,6 @@ jobs:
 
       steps:
       - script: $(_buildScript)
-                -ci
```
Comment (akoeplinger) [inline eng/pipelines/common/templates/build-job.yml]: why?

Comment (kotlarmilos) [reply eng/pipelines/common/templates/build-job.yml]: The `-ci` flag adds a suffix to the version in the maestro build promotion pipeline. Without it, the suffix is `-dev`, which might be more appropriate for standalone experiments.

Comment (akoeplinger) [reply eng/pipelines/common/templates/build-job.yml]: -dev is for local scenarios, we shouldn't be using that in CI

Comment (kotlarmilos) [reply eng/pipelines/common/templates/build-job.yml]: Ok, reverting this change.

**Review by akoeplinger — APPROVED** [review]

**Pull request merged**